src/shared/tests.h

```diff
@@ -84,14 +86,21 @@ static inline int run_test_table(void) {
 
         const TestFunc *t = ALIGN_TO_PTR(__start_SYSTEMD_TEST_TABLE, sizeof(TestFunc*));
         while (t < __stop_SYSTEMD_TEST_TABLE) {
-                log_info("/* %s */", t->name);
-
-                if (t->has_ret) {
-                        int r2 = t->f.int_func();
-                        if (r == EXIT_SUCCESS)
-                                r = r2;
-                } else
-                        t->f.void_func();
+
+                if (t->sd_booted && sd_booted() <= 0) {
+                        log_info("/* systemd not booted, skipping %s */", t->name);
+                        if (t->has_ret && r == EXIT_SUCCESS)
+                                r = EXIT_TEST_SKIP;
```

Comment:
Should we return `EXIT_TEST_SKIP` in this case? At least, we did not.

Reply:
The only cases where this ends up happening is for test-cgroup-mask and test-cgroup-unit-default where imho this should be reported. Note that skipped tests are not reported for tests that return void (majority of TEST_REQ_RUNNING_SYSTEMD tests are void ones).

If really desired, I could make this another TEST() macro param (`.skip_silently = true`) or so, or have the logic reverse (`.report_skipped`)...

Reply:
That's ok to change so. But could you mention that in the commit message?